sufia-models/app/models/concerns/sufia/generic_file/csv.rb

```diff
@@ -0,0 +1,23 @@
+module Sufia
+  module GenericFile
+    module CSV
+      def to_csv
+        ::CSV.generate do |csv|
+          csv << csv_terms.map do |term|
+            values = send(term)
+            values = [values].flatten # make sure we have an array
```

Comment:
You can also use `Array(values)` (or `Array.wrap(values)` in Rails environments) to do this in a way that may be more idiomatic.